rfcs/0021-glossary-terms.md

```diff
@@ -0,0 +1,123 @@
+
+**Alternatives:**
+- External transition - maybe this is more precise, but has the heavy lifting of having to explain a new term for a mostly similar concept. Additionally, the word transition is very similar to transaction, causing confusion at times.
+
+### User Transaction
+**Concept:** A transaction issued by a user - currently a payment or a delegation change
+
+**Rationale:** This term clearly describes the concept, and leaves room for further types of user issued transactions.
+
+**Downsides:** This is a subset of "Transactions", so it is a bit redundant.
+
+**Usage:** There are three types of transactions in the Coda network currently - user transactions, fee transfers, and coinbases.
+
+**Alternatives:**
+- User command - this term is confusing, especially in  docs, as it can be conflated with a CLI command that a user issues.
+
+### Snark worker / Snarker
+**Concept:** A Coda node that produces SNARKs to compress transactions (and in the future, blocks) in the network.
+
+**Rationale:** The community likes SNARKs and it is a point of differentiation for Coda, so snarker is an engaging term.
+
+**Downsides:** Gets invalidated if Coda switches to another type of zero-knowledge proof. NOTE - There is some concern about using the progressive tense of "snark" as a verb, eg. "snarking" - as this can be confused with staking. However, if the community enjoys "snarking", it  would make sense to  continue using it as a verb.
+
+**Usage:** 
+- Anyone can join the coda network and become a snark worker.
+- Snark workers help compress the blockchain by generating SNARKs.
+
+**Alternatives:**
+- Compressor - this term was considered initially, but given the community excitement about SNARKs, it makes sense to include the more specific term in the lexicon.
+
+### Full node
+**Concept:** A Coda node that is able to verify the state of the network - however it may need to request paths for its accounts from other nodes that have all the accounts state.
+
+**Rationale:** By calling this type of node a full node, Coda distinguishes itself from other networks, as all nodes are technically full nodes if they can verify SNARKs - these nodes are not required to trust other nodes.
+
+**Downsides:** Because these nodes still need to request state from other nodes, there is an argument to be made that they are not full nodes. Furthermore, Bitcoin classifies full nodes as nodes that "download every block and transaction and check them against Bitcoin's consensus rules." Therefore, there may be some community pushback against calling these nodes full nodes, even though they are trustless.
+
+**Usage:** On the Coda network, even phones can run full nodes!
+
+**Alternatives:**
+- Fully verifying nodes - more accurate, but a mouthful, and have to create a new term.
+- Trustless nodes - again, some lifting required.
+
+
+### Coda vs coda vs CODA
+**Concept:** "Coda" is the network / protocol. "coda" is the token. "CODA" is the ticker symbol.
```

Comment:
Sometimes we use the phrase "Coda Protocol". Should we try to only use the word "Coda" instead?

Reply:
I think it's fine to be more specific when needed. The way I use it is:
- Coda Protocol: when I'm talking about the uninstantiated network, and all the design decisions around the protocol
- Coda network: referring to the live network, and in the context of sending transaction - it's always on the Coda network
- Coda: the brand, the identity, etc

Can add this and elaborate if you think it'd be useful?

Reply:
I think it's useful, but not strictly necessary for the public definition since there is less scope to how the word "Coda" can be applied from that angle.